src/main/java/org/broadinstitute/hellbender/tools/exome/AllelicCapSegger.java

```diff
@@ -0,0 +1,207 @@
+            shortName = SIGMA_THRESHOLD_ALLELE_FRACTION_SHORT_NAME,
+            optional = true
+    )
+    protected double sigmaThresholdAlleleFraction = 2;
+
+    @Override
+    protected Object doWork() {
+        if (sampleName == null) {
+            sampleName = outputPrefix;
+        }
+
+        logger.info("Starting workflow for sample " + sampleName + "...");
+
+        logger.info("Loading input files...");
+        //make Genome from input target coverages and SNP counts
+        final Genome genome = new Genome(targetCoveragesFile, snpCountsFile, sampleName);
+        //load target-coverage segments from input file and fix up start breakpoints
+        final List<SimpleInterval> targetSegmentsUnfixed =
+                SegmentUtils.readIntervalsFromSegmentFile(targetSegmentsFile);
+        final List<SimpleInterval> targetSegments =
+                SegmentUtils.fixTargetSegmentStarts(targetSegmentsUnfixed, genome.getTargets());
+
+        //segment SNPs on observed log_2 minor allele fraction
+        logger.info("Performing SNP segmentation...");
+        final File snpSegmentFile = new File(outputPrefix + "-" + SNP_MAF_SEG_FILE_TAG + ".seg");
+        SNPSegmenter.writeSegmentFile(genome.getSNPs(), sampleName, snpSegmentFile);
+        final List<SimpleInterval> snpSegments =
+                SegmentUtils.readIntervalsFromSegmentFile(snpSegmentFile);
+
+        //combine SNP and target-coverage segments
+        logger.info("Combining SNP and target-coverage segments...");
+        final List<SimpleInterval> unionedSegments =
+                SegmentUtils.unionSegments(targetSegments, snpSegments, genome);
+        final File unionedSegmentsFile = new File(outputPrefix + "-" + UNION_SEG_FILE_TAG + ".seg");
+        SegmentUtils.writeSegmentFileWithNumTargetsAndNumSNPs(unionedSegmentsFile, unionedSegments, genome);
+
+        //small-segment merging (note that X and Y are always small segments and dropped, since GATK CNV drops them)
+        logger.info("Merging small segments...");
+        final SegmentedModel segmentedModelWithSmallSegments = new SegmentedModel(unionedSegments, genome);
+        final SegmentedModel segmentedModel =
+                segmentedModelWithSmallSegments.mergeSmallSegments(smallSegmentTargetNumberThreshold);
+        final File segmentedModelFile = new File(outputPrefix + "-" + SMALL_MERGED_SEG_FILE_TAG + ".seg");
+        segmentedModel.writeSegmentFileWithNumTargetsAndNumSNPs(segmentedModelFile);
+
+        //initial model fitting
+        logger.info("Fitting initial model...");
+        final ACSModeller modeller = new ACSModeller(segmentedModel, outputPrefix,
+                numSamplesCopyRatio, numBurnInCopyRatio, numSamplesAlleleFraction, numBurnInAlleleFraction);
+
+        //similar-segment merging
+        logger.info("Merging similar segments...");
+        modeller.mergeSimilarSegments(sigmaThresholdCopyRatio, sigmaThresholdAlleleFraction,
```

Comment:
It deserves a comment that output of segment files (including the final one) is a side effect of `mergeSimilaregments` since output is explicit up to this point.